lib/fileset/default.nix

```diff
@@ -763,6 +763,14 @@ in {
     Create a file set containing all [Git-tracked files](https://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository) in a repository.
     The first argument allows configuration with an attribute set,
     while the second argument is the path to the Git working tree.
+
+    Git repositories can also be fetched as a [Nix store path](https://nixos.org/manual/nix/stable/store/store-path.html#store-path)
+    using a Git tree fetcher like `builtins.fetchGit` or `pkgs.fetchgit` (without `leaveDotGit`),
+    for the purposes of so-called out-of-tree building.
+    If such a fetched Git repository uses `gitTrackedWith` on a local path and is evaluated using `import`,
+    the function returns the Git-tracked files even in such a case.
+    So this function doesn't interfere with out-of-tree building.
```

Comment:
This paragraph doesn't feel relatable to someone who's writing a source filter for their project, because they're not using `fetchGit`. I think what we're trying to explain here is that `gitTracked` tries to do the right thing if what's a repo for them is filtered away by someone else. `gitTracked` can only assume that this someone else fetched it the right way, and that's relevant, because `gitTracked` would seem involved in potential problems that are beyond its power.
(Sorry, no time for a suggestion rn)

Reply:
```suggestion
    `gitTrackedWith` does not perform any filtering when the path is a [Nix store path](https://nixos.org/manual/nix/stable/store/store-path.html#store-path) and not a repository.
    In this way, it accommodates the use case where the expression that makes the `gitTracked` call does not reside in an actual git repository anymore, and has presumably already been fetched in a way that excludes untracked files.
    Fetchers that are expected to implement such equivalent behavior include `builtins.fetchGit`, `builtins.fetchTree` (experimental), and `pkgs.fetchgit` when used without `leaveDotGit`.

```

-----

> so-called

Can do without the jargon.

Reply:
The last sentence feels a bit like fetchers need to implement support for this explicitly, how about just:

> Fetchers with such behavior include `builtins.fetchGit`, `builtins.fetchTree` (experimental), and `pkgs.fetchgit` when used without `leaveDotGit`.

Reply:
Yeah sgtm.

I wish we could just claim equivalence, but for example `fetchGit` implements `export-ignore` by default for historic reasons.

Reply:
Oh the "default" part is WIP actually...